tools/acrntrace/README.rst

```diff
@@ -12,6 +12,67 @@ A ``scripts`` directory includes scripts to analyze the trace data.
 Usage
 *****
 
+The ``acrntrace`` tool runs on the Service OS (SOS) to capture trace data and
+output to trace file under ``/tmp/acrntrace`` with raw (binary) data format.
+
+Options:
+
+-h			print this message
+-t period		specify polling interval in milliseconds [1-999]
+-c			clear the buffered old data
+
+The ``acrntrace_format.py`` is a offline tool for parsing trace data (as output
+by acrntrace) to human-readable formats based on given format.
+
+Here's an explanation of the tool's parameters:
+
+.. code-block:: none
+
+   acrntrace_format.py [options] [formats] [trace_data]
+
+Options:
+
+-h    print this message
+
+- A *formats* file specifies the rules to reformat the *trace_data* collected by
+  ``acrntrace`` into a human-readable text form.  The rules in this file are of
+  the form::
+
+    event_id  text_format_string
+
+  The text_format_string may include format specifiers, such as
+  ``%(cpu)d``, ``%(tsc)d``, ``%(event)d``, ``%(1)d``, and ``%(2)d``.
+  The 'd' format specifier outputs in decimal, alternatively 'x' will
+  output in hexadecimal and 'o' will output in octal.
+
+  These respectively correspond to the CPU number (cpu), timestamp
+  counter (tsc), event ID (event) and the data logged in the trace file.
+  There can be only one such rule for each type of event.
+
+  A pre-defined formats file is available in tools/acrntrace/scripts/formats
+  in acrn source tree
+
+The ``acrnalyze.py`` is a offline tool to analyze trace data (as output by
+acrntrace) based on given analyser, such as ``vm_exit`` or ``irq``.
+
+Options:
+
+-h                               print this message
+-i, --ifile=[string]             input file name
+-o, --ofile=[string]             output filename
+-f, --frequency=[unsigned int]   TSC frequency in MHz
+--vm_exit                        generate a vm_exit report
+--irq                            generate an IRQ-related report
+
+.. note:: We depend on TSC frequency to do time-based analysis. Please configure
+   the right TSC frequency that acrn runs on. TSC frequency can be obtained
+   from the ACRN console log (calibrate_tsc, tsc_hz=xxx) when the hypervisor boots.
+
+   The tool does not take into account CPU frequency variation that can
+   occur during normal operation (aka CPU throttling) on the processor which
+   doesn't support for invariant TSC. The results may therefore not be
+   completely accurate in that regard.
+
```

Comment:
Let's add an introduction to the steps below:

    Here's a typical use of ``acrntrace`` to capture trace data from the SOS,
    converting the binary data to human-readable form, 
    copying the processed trace data to your linux system, 
    and running the analysis tool.

Reply:
Thanks @dbkinder. Will do.